test/utils/ktesting/kcontext.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ktesting
+
+import (
+	apiextensions "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	_ "k8s.io/component-base/logs/testinit"
+)
+
+// KContext combines [TContext] (and thus [context.Context] + [testing.TB])
+// with client-go client interfaces. The [TContext] is always usable.
+// The client-go interfaces have to be initialized explicitly and will
+// panic when used before that.
+type KContext interface {
+	TContext
+	clientset.Interface
+	dynamic.Interface
+	apiextensions.Interface
+
+	SetClientSet(client clientset.Interface)
+	SetDynamic(client dynamic.Interface)
+	SetAPIExtensions(client apiextensions.Interface)
+}
```

Comment:
I coming to the conclusion that KContext is too clever for its own good...

Let's instead add `[Set]Client/Dynamic/APIExtensions` to `TContext` and remove KContext.

Reply:
I've removed it in the latest push.